### PR TITLE
GHA-integration commit

### PR DESCRIPTION
**Pull request by Agarwal-Shivansh**

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: .github/depandabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  - package-ecosystem: "docker"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (gab-arrobo) [inline .github/depandabot.yml]:
```suggestion
      interval: "weekly"

```

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        uses: fsfe/reuse-action@v2
+
+  fossa-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: FOSSA scan
+        uses: fossa-contrib/fossa-action@v3
+        with:
+          fossa-api-key: 6d304c09a3ec097ba4517724e4a4d17d
```
Comment (gab-arrobo) [inline .github/workflows/master.yml]:
```suggestion
          fossa-api-key: 6d304c09a3ec097ba4517724e4a4d17d

```

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    - dogsled
+    - unconvert
+    - nakedret
+    - exportloopref
+
+issues:
+  exclude-use-default: false
+  exclude:
+    - Error return value of `.*Close` is not checked
+    - "package-comments: should have a package comment"
```
Comment (gab-arrobo) [inline .golangci.yml]:
```suggestion
    - "package-comments: should have a package comment"

```

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          go-version-file: 'go.mod'
+      - name: Build Docker image
+        run: docker build . -f Dockerfile -t ${DOCKER_REPOSITORY}aether-mock-exporter:${ONOS_SDCORE_ADAPTER_VERSION}
+
+  license-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: reuse lint
+        uses: fsfe/reuse-action@v2
```
Comment (gab-arrobo) [inline .github/workflows/master.yml]:
```suggestion
        uses: fsfe/reuse-action@v3
```

**Review by SeanCondon — APPROVED** [review]

**Comment by gab-arrobo** [comment]
retest this please

**Pull request merged**